internals/secrethub/credential_store.go

```diff
@@ -24,13 +23,16 @@ type CredentialConfig interface {
 	ConfigDir() configdir.Dir
 	PassphraseReader() credentials.Reader
 
-	Register(FlagRegisterer)
+	Register(app *cli.App)
 }
 
 // NewCredentialConfig creates a new CredentialConfig.
 func NewCredentialConfig(io ui.IO) CredentialConfig {
+	dir, _ := configdir.Default()
+	c := ConfigDir{Dir: *dir}
 	return &credentialConfig{
-		io: io,
+		configDir: c,
```

Comment:
This change appeared because kingpin executes the set function of each variable that is used for flags. In our case, the `configDir` parameter of the command is used for the `--config-dir` flag. In kingpin, the function `Set()` would be called, which would put its value to the default one (`$HOME/.secrethub`) if the flag is not used and the envar not set. In cobra this is no longer the case for the default value. Therefore, I had to manually initialize at the creation of the command with the default value.

Another way of solving this is to just call the `Set()` function once with no input. This sets the default value to `$HOME/.secrethub`.

Since calling the function seems simpler and no code is duplicated, I changed to the second approach. Let me know what you think.